ingenialink/pdo.py

```diff
@@ -306,107 +321,80 @@ def reset_tpdo_mapping(self) -> None:
             self.write(map_register, 0)
         self._tpdo_maps.clear()
 
-    def add_rpdo_map(self, rpdo_map: RPDOMap) -> None:
-        """Add a new RPDO map into the drive.
-
-        It takes the first available (not mapped yet) RPDO assignment slot of the drive.
-
-        Args:
-            rpdo_map: RPDO map to be added.
+    def map_rpdos(self) -> None:
+        """Map the RPDO registers into the servo drive.
+        It takes the first available RPDO assignment slot of the drive.
 
         Raises:
             ILError: If there are no available PDOs.
         """
-        if len(self._rpdo_maps) + 1 > self.AVAILABLE_PDOS:
-            raise ILError("Could not add a new RPDO map, there are no available PDOs.")
-        self._rpdo_maps.append(rpdo_map)
-        map_index = len(self._rpdo_maps) - 1
-
+        if len(self._rpdo_maps) > self.AVAILABLE_PDOS:
+            raise ILError(
+                f"Could not map the RPDO maps, received {len(self._rpdo_maps)} PDOs and only"
+                " {self.AVAILABLE_PDOS} are available"
+            )
         self.write(self.RPDO_ASSIGN_REGISTER_SUB_IDX_0, len(self._rpdo_maps))
-        self.write(self.RPDO_MAP_REGISTER_SUB_IDX_0[map_index], len(rpdo_map.items))
-        self.write(
-            self.RPDO_MAP_REGISTER_SUB_IDX_1[map_index],
-            rpdo_map.items_mapping.decode("utf-8"),
-            complete_access=True,
-        )
+        custom_map_index = 0
+        rpdo_assigns = b""
+        for rpdo_map in self._rpdo_maps:
```

Comment:
You could do:
`for custom_map_index, rpdo_map in enumerate(self._rpdo_maps):`

Reply:
It is not the same, if a rpdo_mao is read-only, then custom_map_index is not increased

Reply:
This function has 3 steps:

1. Set RPDO assign register size.
2. Fill custom RPDOs to RPOD map registers. That only must be done if it is not read-only PDO.
3. Set all the RPDO maps to RPDO assign register

Reply:
Finally I divide it in two functions